### ISSUE TITLE
[SUGGESTION] AudioAlign (Aurio) algorithms

### ISSUE THREAD

**Issue by MarcoRavich** [alopatindev/sync-audio-tracks#3]
Hi there, your project is very interesting so I stongly suggest you to "keep an eye" on Aurio/AudioAlign by Mario Guggenberger:

https://github.com/protyposis/AudioAlign

EDIT: Just found another interesting resource -> https://github.com/recastrodiaz/align-audio-recordings-julia

Hope that inspires !

**Comment by MarcoRavich** [comment]
Bump.

Since [1.5.1 version of AudioAlign](https://github.com/protyposis/AudioAlign/releases/) was recently released (together with [Aurio 4.1.0](https://github.com/protyposis/Aurio/releases/)), we've doxed many similar open projects - like yours - in our [HyMPS project](https://forart.it/HyMPS/), under the [AUDIO](https://github.com/forart/HyMPS#-) \ [Treatments page](https://github.com/forart/HyMPS/blob/main/Audio/Treatments.md#--) \ [Alignment/Sync subsection](https://github.com/forart/HyMPS/blob/main/Audio/Treatments.md#alignmentsynch-), to (try) push collaboration between them.

Last but not least, let us know how - in your opinion - we could improve categorization/sorting of collected projects in order to push their evolution by favouring collaborations between developers (and not only).

Hope that helps/inspires !

**Comment by alopatindev** [comment]
Thanks for interest in this project, for mentioning it across other audio synchronization tools, for researching similar tools and for the stimulating this/similar open source tools evolution by pushing devs to collaborate! :heart: 

This project is basically just a refactoring/cleanup of the [project](https://web.archive.org/web/20170916044116/https://nerd.mmccoo.com/2017/06/19/automatically-aligning-multiple-videoaudio-clips-in-kdenlive/) [implemented](https://github.com/mmccoo/nerd_mmccoo/blob/master/audio_sync/sync_kdenlive.cxx) by @mmccoo. I'm not currently actively working on this project, so for anyone interested — please feel free to fork, improve and/or create PR.

On this occasion, I'll share my thoughts on this tool, please refer/quote them anywhere if you find them useful.

Originally I was interested only in simplest scenario: short audio tracks synchronization for the same sound recorded using couple of mics. My use case was vlog recording [tool](https://github.com/alopatindev/vlog-toolset#readme) that, besides other stuff, combines audios taken from phone mic + normal high-quality mic. I'm happy with achieved quality of synchronization here.

Synchronization of movie sound (in original language) and dubbed (in other language) sound is also possible, but very limited.

---

However covering more complex scenarios is still interesting problem. Specifically it'd be nice to support
- long audios with unpredictable **frame drops**
- long audios with unmatching average **speed** (often with with **higher pitch** for faster audio).

Such improvements would make it possible to combine
- probably any restored/remastered videos (movies), often released as 4K video and 5.1 sound (usually in original language only)
- and old (dubbed in other languages) audio tracks, taken from non-remastered versions of the same movies.

If anyone will ever achieve this, this would also imply possibility of the following combination:
- **non-vocal surround** sound taken from the remastered 5.1 sound track
- voiced (**dubbed** in other languages) sound tracks from old non-remastered movie
- the remastered/**restored video** itself.

**Comment by MarcoRavich** [comment]
Thanks for your "stimulating" reply, Alexander !

Actually there are only 3 quite-active open projects about a/v syncing:
- @protyposis's [AudioAlign](https://github.com/protyposis/AudioAlign#readme)/[Aurio](https://github.com/protyposis/Aurio#readme);
- @benfmiller's [Audalign](https://github.com/benfmiller/audalign#readme);
- BBC's (@stephenjolly) [audio-offset-finder](https://github.com/bbc/audio-offset-finder#readme).

We believe that you have to post your interesting ideas/needs in their repos (as we already done in [this issue](https://github.com/benfmiller/audalign/issues/39) or @Johndirr - the [Subsync](https://github.com/Johndirr/SubSync#readme) author - done for visualizazions [here](https://github.com/benfmiller/audalign/issues/43), for example).

Since we're trying to favourite knowledge exchanges between devs, we hope for a "collaborative evolution" of open source projects as free (as in freedom) alternative to commercial products.

Thanks again.

**Comment by alopatindev** [comment]
It looks like [not all](https://github.com/protyposis/Aurio/issues/14#issuecomment-1845888291) of these active projects are currently in active search of improvements.

> BBC's (@stephenjolly) audio-offset-finder.

This tool explicitly states "A **simple** tool and library for **finding the offset** of an audio file within another file" in the readme.

I will be posting my ideas on improvements anyway, where I see it appropriate, thanks!

I will most likely be spreading the ideas on my YouTube channel(s) as well, hopefully this will create more interest for FLOSS developers.

**Comment by stephenjolly** [comment]
Yes, if people were interested in building on our project in a way that moved it away from being simple and general purpose, we'd encourage them to do it as a fork (or indeed as a separate project in its own right) rather than add features to the basic tool.  It's great to see these ideas being discussed though.

**Comment by protyposis** [comment]
> It looks like [not all](https://github.com/protyposis/Aurio/issues/14#issuecomment-1845888291) of these active projects are currently in active search of improvements.

To avoid misunderstandings, Aurio/AudioAlign are open to improvements. What I was meaning to say in that linked post is that I'm not going to implement a list of algorithms just because a random person on the internet posts them :)

**Comment by MarcoRavich** [comment]
> It looks like [not all](https://github.com/protyposis/Aurio/issues/14#issuecomment-1845888291) of these active projects are currently in active search of improvements.
>
You're right (as you can see in [this issue](https://github.com/protyposis/AudioAlign/issues/11), for example), anyway we don't think that's so important in the open source scene: as @stephenjolly and @protyposis clearly claimed here, you can always fork their works... BUT we believe that they can surely provide you the "knowledge" on how their code works in order to develop your needs.

Bytheway having developers (and, why not, users) willing to at least discuss about it is a success already.

Last but not least, it could be also interesting to involve people from specialized communities ([example](https://github.com/protyposis/AudioAlign/issues/10)).

Hope that helps.

**Comment by MarcoRavich** [comment]
BREAKING

@alopatindev may @fsinapsi's [FFmatch](https://github.com/fsinapsi/ffmatch) could fit - a part of - your needs ?

Hope thet helps !

**Comment by MarcoRavich** [comment]
Bump.

There are many interesting improvements both in recent [Aurio](https://github.com/protyposis/Aurio?tab=readme-ov-file#aurio-audio-stream-processing--retrieval-for-net) and [AudioAlign](https://github.com/protyposis/AudioAlign?tab=readme-ov-file#audioalign-audio-synchronization-and-analysis-tool) releases, check them out:

- https://github.com/protyposis/Aurio/releases/
- https://github.com/protyposis/AudioAlign/releases

Some end users' (like us) tests have detected malfunctions, omissions and possible improvements to fix/add, but it would be even more interesting to have some "alignment experts" - like you - feedbacks/opinions too:

- https://github.com/protyposis/Aurio/issues
- https://github.com/protyposis/AudioAlign/issues

Thanks in advance !

**Comment by edsonsantoro** [comment]
Thanks @forart for these ions, mentions. Found good apps because of it.

**Comment by MarcoRavich** [comment]
> Thanks @forart for these ions, mentions. Found good apps because of it.
>
Well, our "issues" are aimed precisely at stimulating the exchange of useful informations/knowledges not only between developers, but among (potential) users too.

**Comment by MarcoRavich** [comment]
Bump.

Full collection of related open "resources" is now located here: https://github.com/FORARTfe/HyMPS/blob/main/Audio/Treatments.md#alignmentsynch-

Hope that helps.
